src/libraries/System.Private.CoreLib/src/System/Reflection/AssemblyNameFormatter.cs

```diff
@@ -54,10 +54,10 @@ public static string ComputeDisplayName(string? name, Version? version, string?
 
             if (cultureName != null)
             {
-                if (cultureName == string.Empty)
+                if (cultureName?.Length == 0)
                     cultureName = "neutral";
                 sb.Append(", Culture=");
-                sb.AppendQuoted(cultureName);
+                sb.AppendQuoted(cultureName!);
```

Comment:
Why is a `!` needed here and a `?` needed on line 57 when the `if` check protects it from null?

Reply:
@danmosemsft, I was getting build error on these lines. The build error itself looks strange because these statements are inside an if condition that makes sure the variable isn't null.

Reply:
If you can find a way to make a small repro case, it will either make clear what we're missing, or give us a bug report to open against Roslyn.

Reply:
The current build fails on line 195 (possible null reference).

![Screenshot_2020-05-15-00-47-26-111_com android chrome](https://user-images.githubusercontent.com/31348972/81993325-bd3ca700-9645-11ea-95f1-4e7ef2721650.jpg)

@danmosemsft

Reply:
@stephentoub is that a Roslyn bug or am I missing something?

Reply:
It's not a compiler bug. The problem is the erroneous question mark three lines earlier.

Reply:
Interesting. I could imagine the compiler could reliably and more usefully warn on the unnecessary question mark instead. But it's not unreasonable, and easily fixed.

Reply:
The compiler sees this as the developer saying "even though I know value isn't null here, I'm still going to do a null check for it, so compiler, be warned that I think value could still be null here", at which point the compiler dutifully says "ok, you know best, and since you checked for null and didn't bail out if it was null and then proceeded to dereference value, warning!"